event.go

```diff
@@ -69,7 +69,7 @@ func jobStatisticsFromInternal(stats *jobstats.JobStatistics) *JobStatistics {
 
 // The maximum size of the subscribe channel. Events that would overflow it will
 // be dropped.
-const subscribeChanSize = 100
+const subscribeChanSizeDefault = 50_000
```

Comment:
Okay, so unfortunately, it seems that Go eagerly allocates channel size [1], and this 50k number is probably way more than most uses will ever need. However, we do need a number at least this high for benchmarking (50k is only ~1s worth of benching which you could imagine falling behind).

@bgentry Thoughts on possibly exposing a second `client.Subscribe` variant that allows a channel size to be injected? (Or actually, it might take a `SubscribeConfig` object just for better future compatibility.) Not perfect, but it may be the best option.

I already had to add an internal subscribe variant so I could fix the test case that checked that subscribe channels can overflow.

---

[1] https://stackoverflow.com/questions/65417280/how-and-when-does-go-allocate-memory-for-bounded-queue-channels

Reply:
BTW, I went ahead and did this (add a second `Subscribe` function that takes `SubscribeConfig`). Feels pretty clean IMO and resolves this problem. Also bumped the default size to 1_000 which feels a little more reasonable.